fossunited/chapters/doctype/foss_chapter/test_foss_chapter.py

```diff
@@ -1,8 +1,95 @@
 # Copyright (c) 2023, Frappe x FOSSUnited and Contributors
 # See license.txt
 
+import frappe
+from faker import Faker
 from frappe.tests.utils import FrappeTestCase
 
 
 class TestFOSSChapter(FrappeTestCase):
-    pass
+    def setUp(self):
+        fake = Faker()
+
+        chapter = frappe.get_doc(
+            {
+                "doctype": "FOSS Chapter",
+                "chapter_name": fake.text(max_nb_chars=40),
+                "chapter_type": "FOSS Club",
+                "city": "Pune",
+                "country": "India",
+                "email": fake.email(),
+                "facebook": fake.url(),
+                "instagram": fake.url(),
+                "linkedin": fake.url(),
+                "mastodon": fake.url(),
+                "public_chat_group_url": fake.url(),
+                "state": "Maharashtra",
+                "x": fake.url(),
+            }
+        )
+        chapter.insert()
+
+        if not frappe.db.exists("Role", "Chapter Team Member"):
+            frappe.get_doc({"doctype": "Role", "role_name": "Chapter Team Member"}).insert(ignore_permissions=True)
+        if not frappe.db.exists("Role", "Chapter Lead"):
+            frappe.get_doc({"doctype": "Role", "role_name": "Chapter Lead"}).insert(ignore_permissions=True)
+
+        lead_profile = frappe.get_doc("FOSS User Profile", {"user": "test@example.com"})
+        chapter.append("chapter_members", {"chapter_member": lead_profile.name, "role": "Lead"})
+        chapter.save()
+
+        self.chapter = chapter
+
+    def tearDown(self):
+        frappe.delete_doc("FOSS Chapter", self.chapter.name)
+
+    def test_role_assignment_on_create(self):
+        # Given a chapter
+        chapter = self.chapter
+
+        # When the chapter was created, a lead was assigned to it.
+        # Then the lead should have the role of 'Chapter Lead' and 'Chapter Team Member'
+        user = frappe.db.get_value("FOSS User Profile", chapter.chapter_members[0].chapter_member, "user")
+        has_role = frappe.db.exists("Has Role", {"role": "Chapter Team Member", "parent": user})
```

Comment:
I am a bit confused about 'Has Role' mention in frappe.db.exists() call

Reply:
Has Role is a child table in Frappe which houses different roles which are linked to a parent of `User`. 

If you have a `Chapter Team Member` role. then `Has Role` will have an entry with `parent` as your user.